File: lab 5/lab 5.1(c++)/MyDigitString.cpp
#pragma once
#include <iostream>
#include "MyDigitString.h"
using namespace std;

MyDigitString::MyDigitString(string str):MyString(str)
{
    for (int i = 0; i < StrLength(); i++)
    {
        if (!(this->str[i] >= '0') || !(this->str[i] <= '9'))
        {
            this->str = this->str.erase(i, 1);
            i--;
        }
    }
}

 string MyDigitString::DeleteSymbol(char symbol)
    {
        for (int i = 0; i < StrLength(); i++)
        {
            if (str[i] == symbol)
            {
                str = str.erase(i, 1);
            }
        }
        return str;
    }
 string MyDigitString::Data()
 { 
     return str; 
 }

File: lab 5/lab 5.1(cs)/MyDigitString.cs
using System;

namespace lab_5
{
    class MyDigitString:MyString
    {
        public MyDigitString(string str):base(str)
        {
            for (int i = 0; i < StrLength(); i++)
            {
                if (!(this.str[i] >= '0') || !(this.str[i] <= '9')) 
                {
                    this.str = this.str.Remove(i, 1);
                    i--;
                }
            }
        }
        public string DeleteSymbol(char symbol)
        {
            for (int i = 0; i < StrLength(); i++)
            {
                if (str[i]==symbol)
                {
                    str = str.Remove(i, 1);
                }
            }
            return str;
        }
        public string Data() => str;
    }
}
